sway-lsp/src/core/token.rs

```diff
@@ -91,6 +91,12 @@ pub fn traverse_node(node: AstNode, tokens: &mut Vec<Token>) {
         AstNodeContent::Declaration(dec) => handle_declaration(dec, tokens),
         AstNodeContent::Expression(exp) => handle_expression(exp, tokens),
         AstNodeContent::ImplicitReturnExpression(exp) => handle_expression(exp, tokens),
+        AstNodeContent::WhileLoop(while_loop) => {
+            handle_expression(while_loop.condition, tokens);
+            for node in while_loop.body.contents {
+                traverse_node(node, tokens);
+            }
```

Comment:
Maybe it's worth abstracting this branch into a small `handle_while_loop` function and then calling that here?

Reply:
Yeah good idea, keeps the `traverse_node` function clean. Will push a fix now